### PR TITLE
Feature/checkout page 2

### PR DESCRIPTION
**Pull request by Lorraine-Lyu**
New functionalities:
1. time picker which disables the time when the store is closed
2. subtotal, total and tax
3. add and decrease the number of item in cart

### CODE REVIEW

File: client/src/Pages/User/Cart/CartProducts.js [hunk 1]
```diff
@@ -28,12 +28,17 @@ const CartProduct = ({ product, deleteItem }) => {
   const handleMouseOut = () => {
     setIsMouseOver(false)
   }
+
   const increase = () => {
     setQuantity(quantity + 1)
+    updateQuantity(quantity + 1)
+    deleteItem(date.getTime())
```
Comment (spoken-angus) [inline client/src/Pages/User/Cart/CartProducts.js:35]: sorry why are we deleting the item here? I see you created a dummyDelete function, but why would we need to delete an item if we adjust the quantity?

File: client/src/Pages/User/Cart/index.js [hunk 2]
```diff
@@ -101,20 +192,21 @@ function CartDetail () {
...
+              <p>{currency(totals.subtotal + totals.tax).format()}</p>
             </div>
             <hr className='breakline' />
           </div>
         </div>
 
         <div className='float-cart__footer'>
-          <div
+          <button
+            disabled={cartItems().length == 0 || pickupTime == null}
```
Comment (spoken-angus) [inline client/src/Pages/User/Cart/index.js:203]: I was able to manually enter an invalid time or even use the Now button even though the store should be closed. I could still use the buy button. You might want to either check if pickUp time is valid or change Now is ASAP and disable typing in the dropdown.

**Review by spoken-angus — COMMENTED** [review]
I see you added antd.css. Add package-lock.json and package.json so when people npm installl it will include antd. Besides that, looks really good!

File: client/src/Pages/User/Cart/CartProducts.js (hunk 1 above)
Comment (Lorraine-Lyu) [reply client/src/Pages/User/Cart/CartProducts.js:35]: Oh that's just an alias for dummyDelete. Sorry the dummyDelete function was written by someone else and I just reused it here, the name does not have a good semantics. I just changed the function name here. All it does is to force update the parent element.

**Pull request merged**